pipeline/runtime/runtestsV2.go

```diff
@@ -49,39 +49,46 @@ func executeRunTestsV2Step(ctx context.Context, f RunFunc, r *api.StartStepReque
 	log := logrus.New()
 	log.Out = out
 	optimizationState := types.DISABLED
+	step := toStep(r)
 	agentPaths := make(map[string]string)
 
-	err := downloadJavaAgent(ctx, tmpFilePath, fs, log)
-	if err != nil {
-		return nil, nil, nil, nil, nil, string(optimizationState), fmt.Errorf("failed to download Java agent")
-	}
+	if r.RunTestsV2.IntelligenceMode {
 
-	rubyArtifactDir, err := downloadRubyAgent(ctx, tmpFilePath, fs, log)
-	if err != nil || rubyArtifactDir == "" {
-		return nil, nil, nil, nil, nil, string(optimizationState), fmt.Errorf("failed to download Ruby agent")
-	}
-	agentPaths["ruby"] = rubyArtifactDir
+		err := downloadJavaAgent(ctx, tmpFilePath, fs, log)
+		if err != nil {
+			return nil, nil, nil, nil, nil, string(optimizationState), fmt.Errorf("failed to download Java agent")
+		}
 
-	pythonArtifactDir, err := downloadPythonAgent(ctx, tmpFilePath, fs, log)
-	if err != nil {
-		return nil, nil, nil, nil, nil, string(optimizationState), fmt.Errorf("failed to download Python agent")
-	}
-	agentPaths["python"] = pythonArtifactDir
+		rubyArtifactDir, err := downloadRubyAgent(ctx, tmpFilePath, fs, log)
+		if err != nil || rubyArtifactDir == "" {
+			return nil, nil, nil, nil, nil, string(optimizationState), fmt.Errorf("failed to download Ruby agent")
+		}
+		agentPaths["ruby"] = rubyArtifactDir
 
-	preCmd, filterfilePath, err := getPreCmd(r.WorkingDir, tmpFilePath, fs, log, r.Envs, agentPaths)
-	if err != nil || pythonArtifactDir == "" {
-		return nil, nil, nil, nil, nil, string(optimizationState), fmt.Errorf("failed to set config file or env variable to inject agent, %s", err)
-	}
+		pythonArtifactDir, err := downloadPythonAgent(ctx, tmpFilePath, fs, log)
+		if err != nil {
+			return nil, nil, nil, nil, nil, string(optimizationState), fmt.Errorf("failed to download Python agent")
+		}
+		agentPaths["python"] = pythonArtifactDir
 
-	commands := fmt.Sprintf("%s\n%s", preCmd, r.RunTestsV2.Command[0])
-	step := toStep(r)
-	step.Command = []string{commands}
-	step.Entrypoint = r.RunTestsV2.Entrypoint
-	setTiEnvVariables(step, tiConfig)
-	err = createSelectedTestFile(ctx, fs, step.Name, r.WorkingDir, log, tiConfig, tmpFilePath, r.Envs, &r.RunTestsV2, filterfilePath)
-	if err != nil {
-		return nil, nil, nil, nil, nil, string(optimizationState), fmt.Errorf("error while creating filter file %s", err)
+		preCmd, filterfilePath, err := getPreCmd(r.WorkingDir, tmpFilePath, fs, log, r.Envs, agentPaths)
+		if err != nil || pythonArtifactDir == "" {
+			return nil, nil, nil, nil, nil, string(optimizationState), fmt.Errorf("failed to set config file or env variable to inject agent, %s", err)
+		}
+
+		commands := fmt.Sprintf("%s\n%s", preCmd, r.RunTestsV2.Command[0])
+		step.Command = []string{commands}
+
+		setTiEnvVariables(step, tiConfig)
```

Comment:
Do we need to set TiEnvVariables only if Intelligence mode is on ?  I can see in runTest step or run step it is getting set irrespective of any condition.

Reply:
yes, this should be set always, moved this above